userspace/libsinsp/sinsp_filter_transformer.cpp

```diff
@@ -63,15 +63,16 @@ bool sinsp_filter_transformer::string_transformer(std::vector<extract_value_t>&
 	return true;
 }
 
-bool sinsp_filter_transformer::transform_type(ppm_param_type& t) const {
+bool sinsp_filter_transformer::transform_type(ppm_param_type& t, uint32_t& flags) const {
+	bool is_list = flags & EPF_IS_LIST;
 	switch(m_type) {
 	case FTR_TOUPPER: {
 		switch(t) {
 		case PT_CHARBUF:
 		case PT_FSPATH:
 		case PT_FSRELPATH:
 			// for TOUPPER, the transformed type is the same as the input type
-			return true;
+			return !is_list;
```

Comment:
Is this potentially a breaking change? It might be that we supported `tolower` (and similars) with list-type fields without noticing. Maybe I'm wrong, just checking in.

Reply:
That makes sense, although I never tried it. I would suggest to test and document this behavior if we want it.

Reply:
Ok, I have reverted this change and I'll add a couple of tests.

Reply:
I ended up rewriting a bunch of tests because they couldn't accommodate this scenarios. PTAL.